egs/lre07/v2/local/lre07_eval/score_lre07.v01d.pl

```diff
@@ -0,0 +1,1137 @@
+	    }
+	}
+
+	my $num_seg_lng = scalar( keys %{ $seg_lngs } );
+
+	if ($$open_set) {
+	    $num_seg_lng--;
+	}
+	if ( $num_seg_lng > 0 ) {
+	    $fa_cost->{$tar_lng} /= ( $num_seg_lng - 1 );
+
+	} else {
+	    die ("FATAL ERROR: No segment language found\n");
+	}
+
+	if ($$open_set) {
+	    $cost_oos->{$tar_lng} = $miss_cost->{$tar_lng} + ( $C_FA * (1 - $P_TARGET) * $fa_prob_oos->{$tar_lng} );
+	    $fa_cost->{$tar_lng} = ( $fa_cost->{$tar_lng} * (1 - $P_TARGET - $P_OOS) ) / (1 - $P_TARGET)  + 
+		( $C_FA * $P_OOS * $fa_prob_oos->{$tar_lng} );
+	}
+	$cost_avg->{$tar_lng} = $miss_cost->{$tar_lng} + $fa_cost->{$tar_lng};
+	$$cost_average += $cost_avg->{$tar_lng};
+    }
+
+    my $num_tar_lng = scalar( keys %{ $tar_lngs } );
+
+    if ($num_tar_lng != 0) {
+	$$cost_average /= $num_tar_lng;
+
+    } else {
+	die ("FATAL ERROR: No target language found 4\n");
+    }
+}
+
+sub count_target_trials {
+    my ($trials, $tar_lang) = @_;
+
+    my $count = 0;
+
+    foreach my $trial ( @{ $trials } ) {
+	if ( $trial->target_language eq $tar_lang) {
+	    $count++;
+	}
+    }
+
+    return $count;
+}
+
+sub count_nontarget_trials {
+    my ($trials, $tar_lang, $seg_lang) = @_;
+
+    my $count = 0;
+
+    foreach my $trial ( @{ $trials } ) {
+	my $norm_seg_lng = normalize_language_name( $trial->segment_language, $trial->test );
+	if ( $trial->target_language eq $tar_lang && $norm_seg_lng eq $seg_lang) {
+	    $count++;
+	}
+    }
+
+    return $count;
+}
+
+sub find_target_languages {
+    my ($tar_lngs, $trials) = @_;
+
+    foreach my $trial ( @{ $trials } ) {
+	my $tar_lng = $trial->target_language;
+	$tar_lngs->{$tar_lng} = 0;
+    }
+
+    if ($DEBUG > 1) {
+	foreach my $tar_lng ( sort keys %$tar_lngs ) {
+	    print "find_target_language: $tar_lng\n";
+	}
+    }
+}
+
+sub find_segment_languages {
+    my ($seg_lngs, $trials) = @_;
+
+    foreach my $trial ( @{ $trials } ) {
+	my $seg_lng = normalize_language_name( $trial->segment_language, $trial->test );
+	$seg_lngs->{$seg_lng} = 0;
+    }
+
+    if ($DEBUG > 1) {
+	foreach my $seg_lng ( sort keys %$seg_lngs ) {
+	    print "find_segment_language: $seg_lng\n";
+	}
+    }
+}
+
+sub merge_trials {
+    my ($tar, $non) = @_;
+
+    my @merged_trials = ();
+    foreach my $trial ( @$tar ) {
+	push ( @merged_trials, $trial );
+    }
+
+    foreach my $trial ( @$non ) {
+	push ( @merged_trials, $trial );
+    }
+
+    if ($DEBUG > 1) {
+	foreach my $trial ( @merged_trials ) {
+	    print "merge_trials: " . $trial->test . " " .
+		$trial->target_language . " " .
+		$trial->segment_language . " " .
+		$trial->decision . " " .
+		$trial->score . "\n";
+	}
+    }
+    return \@merged_trials;
+}
+
+sub numerically {
+
+    if ($a->score > $b->score) {
+	return 1;
+
+    } elsif ($a->score < $b->score) {
+	return -1;
+
+    } else {
+	if ($a->type == 2) {
+	    return 1;
+	} elsif ($a->type == 1) {
+	    return -1;
+	}
+    }
+}
```

Comment:
You probably don't need to do that for scripts that we wrote that are likely to be modified by the user.